src/SDKs/Azure.Core/data-plane/Azure.Core.Tests/ClientTestBase.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+using Castle.DynamicProxy;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    [TestFixture(true)]
+    [TestFixture(false)]
+    public class ClientTestBase
+    {
+        private static readonly ProxyGenerator _proxyGenerator = new ProxyGenerator();
+
+        private static readonly UseSyncMethodsInterceptor _interceptor = new UseSyncMethodsInterceptor();
+
+
+        public bool IsAsync { get; }
+
+        public ClientTestBase(bool isAsync)
+        {
+            IsAsync = isAsync;
+        }
+
+        public virtual TClient CreateClient<TClient>(params object[] args) where TClient: class
+        {
+
+            return WrapClient((TClient)Activator.CreateInstance(typeof(TClient), args));
+        }
+
+        public virtual TClient WrapClient<TClient>(TClient client) where TClient: class
```

Comment:
This needs a better name to indicate it's abstracting away the sync/async difference.  Maybe InstrumentForSyncCalls or something similarly obvious?  Random folks jumping into our tests won't have a chance at figuring this out otherwise.

Reply:
InstrumentForSyncCalls seems slightly out of place in the async test too. Maybe we should name the base type more explicit, we already have SyncAsyncPolicyTestBase. What about SyncAsyncClientTestBase?

Reply:
I think `InstrumentForSyncCalls` works well enough because we're only writing Async calls in the test, but I'd even be fine with a name like `InstrumentClient` if you'd prefer that.  `WrapClient` is just a little more vague than I'm comfortable with.  All of the test class names are going to be rough so I'm fine with whichever you like.